### PR TITLE
Add cloudwatch Target

### PR DESCRIPTION
**Pull request by uritau**
## What does this do?
This PR adds a Cloudwatch Target to simplify the usage of [Cloudwatch datasource](https://grafana.com/docs/features/datasources/cloudwatch/).
.
## Why is it a good idea?
This allows users to use Cloudwatch datasource with specific target (limiting and simplifiying the call) and doesn't interferes with anything else on grafanalib. 

## Questions
- There is no test for this set, I will start working on it ASAP.
- I feel that  hardcoding here the  `validNamespaces` and `validRegions` is not the best approach, but I haven't been able to find any AWS API to gather this information without registering in AWS :( so that's why exists the `checkParams`, with false this will not be updated.

## Usage example
```
from grafanalib.core import *
from grafanalib.cloudwatch import CloudwatchTarget


dashboard = Dashboard(
    title="Clowdwatch Stats",
    rows=[
        Row(panels=[
            Graph(
                title="Cloudwatch ASG",
                dataSource='Cloudwatch',
                targets=[
                    CloudwatchTarget(
                        dimensions={
                            "AutoScalingGroupName": "frontend-asg-name",
                        },
                        metricName="CPUUtilization",
                        namespace="AWS/EC2",
                        region="eu-west-1",
                        statistics=["Average"],
                    ),
                ],
                yAxes=[
                    YAxis(format=PERCENT_UNIT_FORMAT),
                    YAxis(format=SHORT_FORMAT),
                ],
            ),
        ]),
    ],
).auto_panel_ids()
```

### CODE REVIEW

**Comment by atreyd** [comment]
When can we expect this feature to be merged. I am looking forward to use this feature for creating the AWS cloudwatch dashboards using grafanalib.

File: tox.ini [hunk 1]
```diff
@@ -7,7 +7,7 @@
 envlist = py27, py35, py37
 
 [testenv]
-commands = pytest --junitxml=test-results/junit-{envname}.xml
+commands = pytest -o junit_family=xunit2 --junitxml=test-results/junit-{envname}.xml
```
Comment (uritau) [inline tox.ini]: Updating to new junit_family, related with: 
https://github.com/pytest-dev/pytest/issues/6179

File: grafanalib/cloudwatch.py [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+    "AWS/Translate",
+    "AWS/TrustedAdvisor",
+    "AWS/NATGateway",
+    "AWS/TransitGateway",
+    "AWS/VPN",
+    "AWS/WorkMail",
+    "AWS/WorkSpaces",
+]
+
+validRegions = [
```
Comment (dholbach) [inline grafanalib/cloudwatch.py]: Would it make sense to use something like https://stackoverflow.com/a/38451512 here?

File: tox.ini (hunk 1 above)
Comment (dholbach) [reply tox.ini]: Can we make this a separate PR maybe? It definitely makes sense:

```console
.tox/py37/lib/python3.7/site-packages/_pytest/junitxml.py:436
  /home/daniel/dev/grafanalib/.tox/py37/lib/python3.7/site-packages/_pytest/junitxml.py:436: PytestDeprecationWarning: The 'junit_family' default value will change to 'xunit2' in pytest 6.0.
  Add 'junit_family=xunit1' to your pytest.ini file to keep the current format in future versions of pytest and silence this warning.
    _issue_warning_captured(deprecated.JUNIT_XML_DEFAULT_FAMILY, config.hook, 2)
```

**Comment by dholbach** [comment]
Just a quick PSA from me: 
1. we're hanging out in [#grafanalib on Slack](https://weave-community.slack.com/archives/grafanalib)
1. we're planning a contributors meeting soon: here's the [link to the doodle](https://doodle.com/poll/hgsp9wmuff7m9bi7) to find a good time - be sure to pick your timezone before submitting ([preliminary agenda](https://docs.google.com/document/d/1JxrSszyPHYhNbJDWYZehRKv6AO4U-zIBhuNmYQVOIHo/edit))

We'd love to see you there.

File: docs/example-cloudwatch.dashboard.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+The graph shows the Average CPU utilization  from an ASG named "frontend-asg"
+"""
+
+from grafanalib.core import *
+from grafanalib.cloudwatch import CloudwatchTarget
+
+
+dashboard = Dashboard(
+    title="Clowdwatch Stats",
```
Comment (matthewmrichter) [inline docs/example-cloudwatch.dashboard.py]: Typo here but it's inconsequential

**Comment by matthewmrichter** [comment]
@uritau - Please review the circlci tests, it looks like you'll need to make some formatting changes to pass the tests.

File: tox.ini (hunk 1 above)
Comment (dholbach) [reply tox.ini:10]: Moved it to https://github.com/weaveworks/grafanalib/pull/236

File: grafanalib/cloudwatch.py [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+    region = attr.ib(default="default")
+    refId = attr.ib(default="")
+    statistics = attr.ib(default=["Average"], validator=instance_of(list))
+    checkParams = attr.ib(default=True, validator=instance_of(bool))
+
+    def to_json_data(self):
+        if self.checkParams:
+            self.__checkParameters()
+
+        return {
```
Comment (kopatsy) [inline grafanalib/cloudwatch.py:41]: `dict.update` returns `None` (https://docs.python.org/3/library/stdtypes.html#dict.update) which means that the target json ends up being `null`.

**Comment by dholbach** [comment]
Thanks again for your PR. It's been a while since we reviewed this. We'll close the PR for now as grafanalib mainline has already moved on a bit. Please reopen the PR if you (or anyone else) wants to move this work forward again. We're also happy to chat with you [on Slack](https://weave-community.slack.com/archives/C9C9K6T4P) if you have any questions.

**Pull request closed without merging**